zerver/management/commands/sync_ldap_user_data.py

```diff
@@ -33,30 +34,27 @@ def sync_ldap_user_data(
 
         if deactivation_protection:
             if not UserProfile.objects.filter(is_bot=False, is_active=True).exists():
-                error_msg = (
-                    "Ldap sync would have deactivated all users. This is most likely due "
-                    + "to a misconfiguration of LDAP settings. Rolling back...\n"
-                    + "Use the --force option if the mass deactivation is intended."
+                raise Exception(
+                    "LDAP sync would have deactivated all users. This is most likely due "
+                    "to a misconfiguration of LDAP settings. Rolling back...\n"
+                    "Use the --force option if the mass deactivation is intended."
                 )
-                logger.error(error_msg)
-                # Raising an exception in this atomic block will rollback the transaction.
-                raise Exception(error_msg)
             for string_id in realms:
                 if not UserProfile.objects.filter(
                     is_bot=False,
                     is_active=True,
                     realm__string_id=string_id,
                     role=UserProfile.ROLE_REALM_OWNER,
                 ).exists():
-                    error_msg = (
-                        "Ldap sync would have deactivated all owners of realm %s. "
-                        + "This is most likely due "
-                        + "to a misconfiguration of LDAP settings. Rolling back...\n"
-                        + "Use the --force option if the mass deactivation is intended."
+                    raise Exception(
+                        f"LDAP sync would have deactivated all owners of realm {string_id}. "
+                        "This is most likely due "
+                        "to a misconfiguration of LDAP settings. Rolling back...\n"
+                        "Use the --force option if the mass deactivation is intended."
                     )
-                    error_msg = error_msg % (string_id,)
-                    logger.error(error_msg)
-                    raise Exception(error_msg)
+    except Exception:
+        logger.error("LDAP sync failed", exc_info=True)
+        raise
```

Comment:
This should possibly just be doing a `raise CommandError`; not sure.  @mateuszmandera thoughts?

Reply:
@timabbott  Hmm, I think `logger.error` is the right thing here, since this command is supposed to be mostly run in a cronjob rather than being called directly?